microprofile/galleon-common/src/main/resources/modules/system/layers/base/io/smallrye/fault-tolerance/main/module.xml

```diff
@@ -40,15 +40,17 @@
         <module name="jakarta.enterprise.api"/>
         <module name="jakarta.enterprise.concurrent.api"/>
 
+        <module name="io.micrometer" optional="true"/>
         <module name="io.smallrye.common.annotation"/>
         <module name="io.smallrye.config"/>
-        <module name="io.smallrye.metrics"/>
+        <module name="io.smallrye.metrics" optional="true"/>
```

Comment:
@rhusar @jasondlee @kabir I think the io.smallrye.metrics module can be removed altogether. The only thing that depends on it non-optionally is the org.wildfly.extension.microprofile.metrics-smallrye module and that dependency is a bug that should be removed. Once that is gone there is no scenario where the module will be provisioned in a slimmed server. If the smallrye deps of MicroProfile FT and MicroProfile RM can both function correctly without io.smallrye.metrics, then there's no reason to keep shipping it.

Reply:
Thanks @bstansberry I have also realized tonight that this should just be gone instead of made optional here. I am pretty sure this is not required. I ll fix this in the morning.

Reply:
@bstansberry @jasondlee @kabir Opened https://issues.redhat.com/browse/WFLY-17806 to track removal and started working on this.

Reply:
PR is open - https://github.com/wildfly/wildfly/pull/16664